src/middlewared/middlewared/alert/source/active_directory.py

```diff
@@ -1,13 +1,53 @@
-from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, FilePresenceAlertSource
+from datetime import timedelta
+import os
+import logging
+from middlewared.alert.base import AlertClass, AlertCategory, Alert, AlertLevel, ThreadedAlertSource
+from middlewared.alert.schedule import IntervalSchedule
 
+log = logging.getLogger("activedirectory_check_alertmod")
 
 class ActiveDirectoryDomainBindAlertClass(AlertClass):
     category = AlertCategory.DIRECTORY_SERVICE
     level = AlertLevel.WARNING
-    title = "ActiveDirectory Did Not Bind to the Domain"
-    text = "ActiveDirectory Did Not bind to the domain."
+    title = "ActiveDirectory Bind Is Not Healthy"
+    text = "Attempt to connect to netlogon share failed with error: %(wberr)s."
 
 
-class ActiveDirectoryDomainBindAlertSource(FilePresenceAlertSource):
-    path = "/tmp/.adalert"
-    klass = ActiveDirectoryDomainBindAlertClass
+class ActiveDirectoryDomainHealthAlertClass(AlertClass):
+    category = AlertCategory.DIRECTORY_SERVICE
+    level = AlertLevel.WARNING
+    title = "ActiveDirectory Domain Validation Failed"
+    text = "Domain validation failed with error: %(verrs)s."
+
+
+class ActiveDirectoryDomainHealthAlertSource(ThreadedAlertSource):
+    schedule = IntervalSchedule(timedelta(hours=24))
```

Comment:
If we run this every 24 hours, can'it it be better idea to use `CronSchedule` instead and run at the same time every day?

Reply:
Good idea.